### PR TITLE
CE-1845 Handle POST requests from Flags edit form

### PR DESCRIPTION
**Pull request by adamkarminski**
@kamilkoterba @Grunny

### CODE REVIEW

File: extensions/wikia/Flags/controllers/FlagsController.class.php [hunk 1]
```diff
@@ -79,24 +83,12 @@ public function getFlagsForPageForEdit() {
...
-		 * 2. Get all flag types for a wikia
-		 */
-		$this->model = new FlagType();
-		$flagTypesForWikia = $this->model->getFlagTypesForWikia( $this->params['wikiId'] );
-
-		/**
-		 * 3. Unite the arrays
+		 * 1. Get all flag types with instances for the page
 		 */
-		$responseData = $flagsForPage + $flagTypesForWikia;
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: Can we explain here why we can combine flags with types. Or change naming.

File: extensions/wikia/Flags/FlagsHelper.class.php [hunk 2]
```diff
@@ -6,6 +6,85 @@
 
 class Helper {
 
+	public function compareDataAndGetFlagsToChange( $currentFlags, $postData ) {
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/FlagsHelper.class.php]: Documentation

File: extensions/wikia/Flags/controllers/FlagsController.class.php [hunk 3]
```diff
@@ -240,4 +284,23 @@ public function removeFlagType() {
...
+		/**
+		 * 2. Get all flag types for a wikia
+		 */
+		$flagTypeModel = new FlagType();
+		$flagTypesForWikia = $flagTypeModel->getFlagTypesForWikia( $wikiId );
+
+		/**
+		 * 3. Return the united arrays
+		 */
+		return $flagsForPage + $flagTypesForWikia;
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: Can we explain here why we can combine flags with types. Or change naming.

**Comment by kamilkoterba** [comment]
Despite from "Can we explain here why we can combine flags with types. Or change naming" comment
LGTM

File: extensions/wikia/Flags/controllers/FlagsController.class.php [hunk 4]
```diff
@@ -146,6 +138,57 @@ public function getFlagsForPage() {
...
+
+		/**
+		 * Get the current status to compare
+		 */
+		$currentFlags = $this->getAllFlagTypes( $this->params['wikiId'], $this->params['pageId'] );
+
+		$helper = new Helper();
+		$flagsToChange = $helper->compareDataAndGetFlagsToChange( $currentFlags, $this->params );
+
+		$this->model = new Flag();
```
Comment (Grunny) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: Is there a reason the model is in a class property? Right now every method that uses it creates a new instance anyway.

File: extensions/wikia/Flags/FlagsHelper.class.php [hunk 5]
```diff
@@ -38,4 +130,8 @@ private function sendGetFlagsForPageRequest( $pageId ) {
 			]
 		)->getData();
 	}
+
+	private function composeInputName( $flagTypeId, $field ) {
+		return self::FLAGS_INPUT_NAME_PREFIX . ":{$flagTypeId}:" . $field;
```
Comment (Grunny) [inline extensions/wikia/Flags/FlagsHelper.class.php]: `$field` can be part of the interpolated string as well.

**Comment by Grunny** [comment]
LGTM.

File: extensions/wikia/Flags/FlagsHelper.class.php [hunk 6]
```diff
@@ -6,6 +6,98 @@
...
+			 * Check only for names of parameters defined in the flag_param_names field.
+			 * This is a protection from users modifying names of parameters in the DOM.
+			 */
+			foreach ( $paramNames as $paramName ) {
+				$key = $this->composeInputName( $flagTypeId, $paramName );
+				if ( isset( $postData[$key] ) ) {
+					/**
+					 * Use a value from the form if it is provided
+					 */
+					$flagFromPost['params'][$paramName] = \Sanitizer::escapeHtmlAllowEntities( $postData[$key] );
```
Comment (Grunny) [inline extensions/wikia/Flags/FlagsHelper.class.php]: Why is this being escaped here, i.e. where is it going from here, and how will this be output in the end? If these parameters end up being parsed as wikitext, they could end up double escaped.

**Comment by Grunny** [comment]
Left some questions, but overall LGTM.

File: extensions/wikia/Flags/controllers/FlagsController.class.php (hunk 4 above)
Comment (adamkarminski) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: Yep, I think that at this point we can easily say that storing a model as the Controller's property is pointless. Thanks!

File: extensions/wikia/Flags/FlagsHelper.class.php (hunk 5 above)
Comment (adamkarminski) [inline extensions/wikia/Flags/FlagsHelper.class.php]: :+1:

File: extensions/wikia/Flags/models/FlagParameter.class.php [hunk 7]
```diff
@@ -36,6 +36,25 @@ public function createParametersForFlag( $flagId, $flagTypeId, $wikiId, $pageId,
 		return $this->status;
 	}
 
+	public function updateParametersForFlag( $flagId, $params ) {
+		$db = $this->getDatabaseForWrite();
+
+		foreach ( $params as $paramName => $paramValue ) {
+			( new \WikiaSQL )
+				->UPDATE( self::FLAGS_PARAMS_TABLE )
+				->SET( $paramName, $paramValue )
```
Comment (Grunny) [inline extensions/wikia/Flags/models/FlagParameter.class.php]: Shouldn't this be `->SET( 'param_name', $paramValue )`?

Comment (adamkarminski) [inline extensions/wikia/Flags/models/FlagParameter.class.php]: Actually, it should be `->SET( 'param_value', $paramValue )` :D

**Pull request merged**